readthedocs/search/views.py

```diff
@@ -178,10 +180,24 @@ def get(self, request):
             role_name=request.GET.get('role_name'),
         )
 
-        results, facets = self._search(
-            user_input=user_input,
-            use_advanced_query=True,
-        )
+        projects = []
+        # If we allow private projects,
+        # we only search on projects the user belongs or have access to.
+        if settings.ALLOW_PRIVATE_REPOS:
+            projects = list(
+                Project.objects.for_user(request.user)
+                .values_list('slug', flat=True)
+            )
+
+        # Make sure we always have projects to filter by if we allow private projects.
+        if settings.ALLOW_PRIVATE_REPOS and not projects:
```

Comment:
💯

Reply:
This should log an error, as well.

Reply:
Not sure about the error, basically this case will be hit by people that don't have projects in .com. Should I log that?

Reply:
I have added a test case for empty projects

Reply:
Hrm, true. I guess it's fine, but I feel like we could hit this logic and not know why in some other behavior. I guess it's fine.